ada_feeding/ada_feeding/behaviors/moveit2/moveit2_execute.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+This module defines the MoveIt2Excute behavior, which uses pymoveit2
+to execute and JointTrajectory.
+"""
+
+# Standard imports
+from typing import Union, Optional
+
+# Third-party imports
+from action_msgs.msg import GoalStatus
+from overrides import override
+from moveit_msgs.msg import MoveItErrorCodes
+import py_trees
+from pymoveit2 import MoveIt2State
+from rclpy.duration import Duration
+from trajectory_msgs.msg import JointTrajectory
+
+# Local imports
+from ada_feeding.helpers import BlackboardKey, get_moveit2_object
+from ada_feeding.behaviors import BlackboardBehavior
+
+
+class MoveIt2Execute(BlackboardBehavior):
+    """
+    Runs moveit2.py execute with the provided
+    JointTrajectory.
+    """
+
+    # pylint: disable=arguments-differ
+    # We *intentionally* violate Liskov Substitution Princple
+    # in that blackboard config (inputs + outputs) are not
+    # meant to be called in a generic setting.
+
+    def blackboard_inputs(
+        self,
+        trajectory: Union[BlackboardKey, Optional[JointTrajectory]],
+        terminate_timeout: Union[BlackboardKey, Duration] = Duration(seconds=10.0),
+        terminate_rate_hz: Union[BlackboardKey, float] = 30.0,
+    ) -> None:
+        """
+        Blackboard Inputs
+
+        Parameters
+        ----------
+        trajectory: JointTrajectory to execute. If None, return SUCCESS immediately.
+        terminate_timeout: How long after a terminate is requested to wait for a
+            response from the MoveIt2 action server.
+        terminate_rate_hz: How often to check whether a terminate request has been
+            processed.
+        """
+        # pylint: disable=unused-argument, duplicate-code
+        # Arguments are handled generically in base class.
+        super().blackboard_inputs(
+            **{key: value for key, value in locals().items() if key != "self"}
+        )
+
+    def blackboard_outputs(
+        self,
+        error_code: Optional[
+            BlackboardKey
+        ] = None,  # Union[MoveItErrorCodes, GoalStatus] == int
+    ) -> None:
+        """
+        Blackboard Outputs
+        By convention (to avoid collisions), avoid non-None default arguments.
+
+        Parameters
+        ----------
+        error_code: specifies the manner of execution (or goal) failure
+                    See actionlib_msgs/GoalStatus and moveit_msgs/MoveItErrorCodes for more info
+                    <0: MoveItErrorCodes Failure
```

Comment:
The error codes are both type int, I'll update.
It's a case where MoveItErrorCodes and GoalStatus are msg objects with constant ints defined inside. 

Basically, the return can be either, which works because when there's an error GoalStatus reports a positive number and Move it reports a negative number.

Reply:
I don’t think they are both interested. To get to the Int, users will have to call `error_code.status` if it is a GoalStatus, or `error_code.Val` if it is a MoveItErrorCode. Hence, users are already conditioning on the type of error code.

Reply:
EDIT: I think we are saying the same thing. I just felt like your comment made it more confusing, but I understand your perspective that pointing out in the commend that one case is always positive and the other is negative can be helpful. I think just clarifying the comment should resolve this.

Reply:
Just made sure that error_code is always the full type and not an int (I made that mistake once by returning `MoveItErrorCodes.SUCCESS`)